src/Control/ContentNegotiator.php

```diff
@@ -84,12 +91,37 @@ public static function enabled_for($response)
             return false;
         }
 
-        if (static::config()->get('enabled')) {
+        if (ContentNegotiator::getEnabled()) {
             return true;
         } else {
             return (substr($response->getBody(), 0, 5) == '<' . '?xml');
         }
     }
+    
+    /**
+     * Gets the current enabled status, if it is not set this will fallback to config
+     *
+     * @return bool
+     */
+    public static function getEnabled()
+    {
+        if (!isset(self::$current_enabled)) {
+            self::$current_enabled = ContentNegotiator::config()->get('enabled');
```

Comment:
I want to change my mind about something; I think we should not modify state inside the getters (even though i18n does this). The justification for i18n was due to performance and the fact that i18n::get_theme() is called hundreds of times.

For less-frequently accessed properties, shall we use
```
if (!isset(self::$current_enabled)) {
  return ContentNegotiator::config()->get('enabled');
}
```

Instead?

Reply:
hmm, I kind of agree with your proposal about the less frequently accessed properties, but I'm wary of setting another paradigm for them just because of the fact that they're not used frequently
reference:
```
if (!isset(self::$current_enabled)) {
  return ContentNegotiator::config()->get('enabled');
}
return self::$current_enabled;
```
would you be opposed to updating `i18n::get_locale()` to follow?
a config read should be cheap right? and the whole purpose of this pull request is to remove set/merge calls to the config in Core

Reply:
We've agreed to update i18n as well so we can be consistent with this suggestion - in such a way that it doesn't use `Config_ForClass` which was the root cause of the slow down for i18n